src/nfa/nfa.cc

```diff
@@ -1281,31 +1281,70 @@ Nfa Mata::Nfa::minimize(
 	return algo(aut);
 }
 
-void Nfa::print_to_DOT(std::ostream &outputStream) const {
-    outputStream << "digraph finiteAutomaton {" << std::endl
+std::string Nfa::print_to_DOT() const {
+    std::stringstream output;
+    print_to_DOT(output);
+    return output.str();
+}
+
+void Nfa::print_to_DOT(std::ostream &output) const {
+    output << "digraph finiteAutomaton {" << std::endl
                  << "node [shape=circle];" << std::endl;
 
-    for (State finalState: final) {
-        outputStream << finalState << " [shape=doublecircle];" << std::endl;
+    for (State final_state: final) {
+        output << final_state << " [shape=doublecircle];" << std::endl;
     }
 
     const size_t delta_size = delta.num_of_states();
-    for (State s = 0; s != delta_size; ++s) {
-        for (const Move &t: delta[s]) {
-            outputStream << s << " -> {";
-            for (State sTo: t.targets) {
-                outputStream << sTo << " ";
+    for (State source = 0; source != delta_size; ++source) {
+        for (const Move &tran: delta[source]) {
+            output << source << " -> {";
+            for (State target: tran.targets) {
+                output << target << " ";
             }
-            outputStream << "} [label=" << t.symbol << "];" << std::endl;
+            output << "} [label=" << tran.symbol << "];" << std::endl;
         }
     }
 
-    outputStream << "node [shape=none, label=\"\"];" << std::endl;
-    for (State initialState: initial) {
-        outputStream << "i" << initialState << " -> " << initialState << ";" << std::endl;
+    output << "node [shape=none, label=\"\"];" << std::endl;
+    for (State init_state: initial) {
+        output << "i" << init_state << " -> " << init_state << ";" << std::endl;
     }
 
-    outputStream << "}" << std::endl;
+    output << "}" << std::endl;
+}
+
+std::string Nfa::print_to_mata() const {
+    std::stringstream output;
+    print_to_mata(output);
+    return output.str();
+}
+
+void Nfa::print_to_mata(std::ostream &output) const {
+    output << "@NFA-explicit" << std::endl
+           << "%Alphabet-auto" << std::endl;
+           // TODO should be this, but we cannot parse %Alphabet-numbers yet
+           //<< "%Alphabet-numbers" << std::endl;
```

Comment:
What do you need for this? Could we modify alphabets to be able to do so? I will have to check what `@Alphabet-numbers` is. Is `@Alphabet-numbers` even described in AUTOMATAFORMAT.md? I cannot find it there. Maybe we should have the alphabet names used in our format beside its descriptions.

Reply:
It is mentioned under `Distinguishing between states, symbols, and nodes`. For it to work, we would have to modify the parser, currently it does not support it at all. This is why I need to use `IntAlphabet` if I want to parse output of this function, so that it creates the same symbols.

Reply:
Got it. Thank you for the explanation. Seems good to me for now, then.